sklearn/datasets/value_dropper.py

```diff
@@ -0,0 +1,333 @@
+# Author : Raghav RV <rvraghav93@gmail.com>
+#
+# Licence : BSD 3 clause
+
+import numpy as np
+import numbers
+
+from sklearn.utils import check_array
+from sklearn.utils import check_random_state
+from sklearn.utils.multiclass import type_of_target
+
+from sklearn.base import TransformerMixin
+from sklearn.preprocessing import LabelEncoder
+
+
+__all__ = ["ValueDropper"]
+
+
+class ValueDropper(TransformerMixin):
+    """Artificially insert NMAR or MCAR missing values into data.
+
+    Where,
+
+    NMAR/MNAR - Not Missing At Random / Missing Not At Random
+        When the missingness is correlated with the class classes in the
+        target (y) (and hence informative).
+
+    MCAR - Missing Completely At Random
+        When the missingness is completely random (and hence uninformative).
+
+    If the missing type is NMAR, a ``missing_proba`` parameter can be passed
+    to drop values conforming to the given drop-probabilities.
+
+
+    Parameters
+    ----------
+
+    missing_values : {"NaN" (or np.nan) | int}, default "NaN"
```

Comment:
In fact in float you have `NaN` specially for that purpose and none will use another value for that purpose. `NaN` doesn't exist in int so you have to choose a specific value for that purpose and that value depends of your data, ...